average.js

```diff
@@ -1,6 +1,15 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  let ans = 0;//initialize answer
+  let len = numbers.length;//length of array
+  numbers.map((num,i)=>{
+    if(!isNaN(num)){
+      ans += num; //add value to ans if it is a number
```

Comment:
Nice implementation in a single pass. Using a single loop to count and summate is good in many contexts.
Consider the other option as well: If the counting and summation are done in different loops, we could parallelize them, resulting in faster execution for large data-sets.

Reply:
I have split up the loops for the calculation of counting NaN values and summing up the average to parallelize the function.

Please let me know if this is what the requirement was or if any changes are required.